acceptance/acceptance_test.go

```diff
@@ -0,0 +1,275 @@
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return ctx, fmt.Errorf("running ec validate input: %w\n%s", err, stderr.String())
+	}
+
+	ts.cmd = cmd
+
+	var r report
+	if err := json.Unmarshal(stdout.Bytes(), &r); err != nil {
+		return ctx, fmt.Errorf("unmarshalling report: %w", err)
+	}
+	ts.report = r
+
+	return setTestState(ctx, ts), nil
+}
+
+func thereShouldBeNoViolationsInTheResult(ctx context.Context) error {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return fmt.Errorf("reading test state: %w", err)
+	}
+
+	for _, filepath := range ts.report.FilePaths {
+		if len(filepath.Violations) != 0 {
+			return errors.New(prettifyResults("expected no violations, got:", filepath.Violations))
+		}
+	}
+
+	return nil
+}
+
+func thereShouldBeNoWarningsInTheResult(ctx context.Context) error {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return fmt.Errorf("reading test state: %w", err)
+	}
+
+	for _, filepath := range ts.report.FilePaths {
+		if len(filepath.Warnings) != 0 {
+			return errors.New(prettifyResults("expected no warnings, got:", filepath.Warnings))
+		}
+	}
+
+	return nil
+}
+
+func prettifyResults(msg string, results []result) string {
+	for _, violation := range results {
+		code := violation.Metadata["code"].(string)
+		msg += fmt.Sprintf("\n\t%s:\t%s", code, violation.Message)
+	}
+	return msg
+}
+
+func replaceVariables(content string, variables map[string]string) string {
+	for name, value := range variables {
+		re := regexp.MustCompile(`\$` + name + `\b`)
+		content = re.ReplaceAllString(content, value)
+	}
+	return content
+}
+
+func setupScenario(ctx context.Context, sc *godog.Scenario) (context.Context, error) {
+	tempDir, err := os.MkdirTemp("", "ec-policies-")
+	if err != nil {
+		return ctx, fmt.Errorf("setting up scenario: %w", err)
+	}
+
+	gitroot, err := guessGitRoot()
+	if err != nil {
+		return ctx, fmt.Errorf("getting gitroot: %w", err)
+	}
+
+	ts := testState{
+		cliPath: filepath.Join(gitroot, "acceptance/bin/ec"),
+		tempDir: tempDir,
+		variables: map[string]string{
+			"GITROOT": gitroot,
+		},
+	}
+
+	return setTestState(ctx, ts), nil
+}
+
+func tearDownScenario(ctx context.Context, sc *godog.Scenario, _ error) (context.Context, error) {
+	// Purposely ignore errors here to prevent a tear down error to mask a test error.
+	ts, _ := getTestState(ctx)
+
+	if ts.tempDir != "" {
+		_ = os.RemoveAll(ts.tempDir)
+	}
+	return ctx, nil
+}
+
+func getTestState(ctx context.Context) (testState, error) {
+	ts, ok := ctx.Value(testStateKey{}).(testState)
+	if !ok {
+		return testState{}, errors.New("test state not set")
+	}
+	return ts, nil
+}
+
+func setTestState(ctx context.Context, ts testState) context.Context {
+	return context.WithValue(ctx, testStateKey{}, ts)
+}
+
+// guessGitRoot looks for a directory containing a .git directory. It starts from the current
+// working dirctory and walks up the directory tree.
+func guessGitRoot() (string, error) {
```

Comment:
I don't think this is needed, tests are guaranteed to run with CWD of `$GITROOT/acceptance`, so we can assume `GITROOT=..`

Reply:
Nice! I didn't know that.

Reply:
https://github.com/enterprise-contract/ec-policies/pull/866